src/components/search/data/searchLoader.js

```diff
@@ -18,11 +19,19 @@ export default function makeSearchLoader(queryClient) {
       authenticatedUser,
       enterpriseSlug,
     });
-    const searchData = [
-      queryClient.ensureQueryData(
-        queryAcademiesList(enterpriseCustomer.uuid),
-      ),
-    ];
+    const academies = await queryClient.ensureQueryData(queryAcademiesList(enterpriseCustomer.uuid));
+
+    if (enterpriseCustomer.enableOneAcademy) {
+      if (academies.length === 1) {
```

Comment:
nit: could these conditionals be combined into one to avoid nested conditionals here? 

```jsx
if (enterpriseCustomer.enableOneAcademy && academies.length === 1) {
  return redirect(redirectPath);
}
```